### DISCUSSION TITLE
Mullvad not in raspap.sudoers after update?

### DISCUSSION THREAD

**Discussion by E53L** [RaspAP/raspap-webgui#1645] [Q&A] [answered]
Hi.

It seems like after installing an update my Mullvad GUI is broken:
![gui_mv](https://github.com/user-attachments/assets/c0f880aa-31db-4cd6-b10c-fdd150f6d36e)

Apparently Mullvad works just fine in the background, though. I guess I need to know the commands that execute in the initial setup of RaspAP when you choose to install the Mullvad GUI, but I can't find those in the "manual installation" section.

If I need to add Mullvad manually to raspap.sudoers I could aslo use some help, as my Linux skills are unfortunately still limited.

Thank you in advance

**Comment by billz** [comment, marked as answer]
VPN providers are [defined here](https://github.com/RaspAP/raspap-webgui/blob/master/config/vpn-providers.json). Execute `sudo visudo /etc/sudoers.d/090_raspap` and append the following:

```
www-data ALL=(ALL) NOPASSWD:/usr/bin/mullvad *
```

Save the file, exit and load the VPN providers UI again. The sudoers file is modified at install time when the VPN provider option is selected. Running the update installs a new sudoers, but `/etc/raspap/provider.ini` should be parsed so that this line is appended. This will need fixing.

**Reply by E53L** [reply]
Yesss, it works! Thank you!
